### DISCUSSION TITLE
Prevent going back to previous survey box on the same screen

### DISCUSSION THREAD

**Discussion by belginunal** [jspsych/jsPsych#1484] [Q&A] [answered]
Hi,

I have an experiment where people are given three boxes on the screen, and they have to type in their responses in a certain order.
Is there a way to prevent people from going back to the previous box after they clicked on the next box (while keeping these boxes on the same screen)? Survey-text does not seem to have that option, so I was wondering if there is a workaround.

I would really appreciate any advice. Thank you!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @belginunal, to do this I think you'd need a function that is called whenever the textbox focus changes from being in focus (i.e. currently accepting input) to not being in focus anymore. You can do this by adding a [`onblur`](https://www.w3schools.com/jsref/event_onblur.asp) event listener to all of the text boxes. In that function, you can [disable the textbox](https://www.w3schools.com/jsref/prop_text_disabled.asp) so that the input can't be changed anymore. [This page](https://javascript.info/focus-blur) also has a nice example of the onfocus and onblur events.

**Reply by belginunal** [reply]
Thank you so so much for the response @becky-gilbert! As you suggested, the code below does what I need: 
```
<!DOCTYPE html>
<html>
<body>

<input type="text" id="name">

<script>
document.getElementById("name").addEventListener("blur", myFunction);

function myFunction() {
  document.getElementById("name").disabled = true;
}
</script>

</body>
</html>

```
But, I could not integrate this into a survey-text type variable. I apologize in advance for my ignorant question, I am new to Javascript.

**Reply by becky-gilbert** [reply]
Hi @belginunal, sorry for the long delay getting back to you. I'm not sure if this is still useful, but thought I'd share my solution anyway. One way to do this is to find all of the text boxes on the page using the `querySelectorAll` function. Then you can use `forEach` to add a `blur` event listener that disables each text box after a blur event occurs (i.e. the same thing that you posted above). You can add this bit of code to the `survey-text` plugin after the HTML has been added to the page, e.g. around [line 180](https://github.com/jspsych/jsPsych/blob/bf9dec250b7fcd038171cb3a1aad5647d5bb67e7/plugins/jspsych-survey-text.js#L180) in the v6.3.0 plugin version.
```js
// add blur event listeners to all text boxes to disable the text box after the cursor is moved out of it
document.querySelectorAll('textarea, input').forEach(function(textbox) {
    textbox.addEventListener('blur', function() { this.disabled = true; });
});
```
Note that this will always disable the text box after the cursor has moved out of it, even if the participant hasn't entered a response. If you want to change that, you'd need to do some validation in the `blur` event listener. For instance, you could check the value of the text box, and then only disable it if it's not empty, or if it meets some other criteria.

**Reply by belginunal** [reply]
This is super useful! Thank you so much @becky-gilbert!!!
